ldms/src/sampler/slingshot_info/slingshot_info.c

```diff
@@ -386,6 +388,10 @@ static int sample(struct ldmsd_sampler *self)
 static void term(struct ldmsd_plugin *self)
 {
         log_fn(LDMSD_LDEBUG, SAMP" term() called\n");
+        base_del(sampler_base);
```

Comment:
`base_del()` frees `rec_def` when it deletes the schema by calling `ldms_schema_delete()`, so there is no need to call `ldms_record_delete(rec_def)`.`rec_def` is a part of the schema when `ldms_schema_record_add()` is called, so `ldms_schema_delete()` frees `rec_def` altogether. 
However, `base_del()` doesn't delete the set. You have to delete the set here.

The procedure here should be.
```
base_set_delete(sampler_base);
base_del(sampler_base);
sampler_base = NULL;
rec_def = NULL;
```

Reply:
OK, thanks.

Reply:
Oh! In that case, there is no real reason for me to even have rec_def as global scope. That is even better.

Reply:
Addressed in the latest push.